app/models/case/sar/offender.rb

```diff
@@ -556,4 +556,11 @@ def set_number
                     next_number
                   end
   end
+
+  def set_deadlines
+    super
+    if rejected?
+      self.external_deadline = @deadline_calculator.days_after(90, received_date)
```

Comment:
Try to avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)).

You could create a constant in the class for this value.